blurring_as_a_service/endpoint/components/score.py

```diff
@@ -40,36 +42,149 @@ def init():
     You can write the logic here to perform init operations like caching the model in memory
     """
     global model
-    # AZUREML_MODEL_DIR is an environment variable created during deployment.
-    # It is the path to the model folder (./azureml-models/$MODEL_NAME/$VERSION)
-    # Please provide your model's folder name if there is one
-    model_path = os.path.join(
-        os.getenv("AZUREML_MODEL_DIR"), "yolov8m_1280_v2.2_curious_hill_12.pt"
-    )
-    print("Init started")
-    model = YOLO(
-        model=model_path,
-        task="detect",
-    )
-    print("Init complete")
+    global settings
+    global logger
+
+    logger.info("Init started")
+    try:
+        # AZUREML_MODEL_DIR is an environment variable created during deployment.
+        # It is the path to the model folder (./azureml-models/$MODEL_NAME/$VERSION)
+        # Please provide your model's folder name if there is one
+        model_path = os.path.join(
+            os.getenv("AZUREML_MODEL_DIR"), "yolov8m_1280_v2.2_curious_hill_12.pt"
+        )
+        if not os.path.exists(model_path):
+            logger.error(f"Model file not found at {model_path}")
+            raise FileNotFoundError(f"Model file not found: {model_path}")
+        if settings is None:
+            raise RuntimeError("Configuration settings could not be loaded.")
+
+        logger.info(f"Loading model from: {model_path}")
+        model = YOLO(
+            model=model_path,
+            task="detect",
+        )
+        logger.info("Init complete")
+    except FileNotFoundError as e:
+        logger.error(f"Initialization failed: Model file not found. {e}")
+        raise
+    except EnvironmentError as e:
+        logger.error(f"Initialization failed: Environment configuration error. {e}")
+        raise
+    except RuntimeError as e:
+        logger.error(f"Initialization failed: Critical runtime error. {e}")
+        raise
+    except Exception as e:
+        logger.error(f"An unexpected error occurred during init: {e}", exc_info=True)
+        raise
 
 
 def run(raw_data):
+    """
+    This function is called for every invocation of the endpoint.
+    Parameters:
+        raw_data (str): The raw request data in JSON format.
```

Comment:
It would be nice to have a decsription of the expected JSON format, e.g. which fields are required.